aws_topology/stackstate_checks/aws_topology/resources/s3.py

```diff
@@ -55,30 +68,35 @@ def collect_buckets(self):
         ]:
             yield bucket
 
-    def process_all(self, filter=None):
+    @set_required_access_v2("s3:ListAllMyBuckets")
+    def process_buckets(self):
         # TODO buckets should only be fetched for global OR filtered by LocationConstraint
-        if not filter or 'buckets' in filter:
```

Comment:
I think we still need the filter because there will be a lot more from this API later? Or do you want to add it later?

Reply:
This line is still in there, just got moved to another place - check the diff